### DISCUSSION TITLE
LDAP federation without LDAP auth

### DISCUSSION THREAD

**Discussion by gedpal** [keycloak/keycloak#31139] [Ideas]
Hello,
I need to realize a solution using Keycloak to sync users to local KC user DB but preventing to use LDAP pass authentication (security requirement). I found out that LDAP sync mode: UNSYNCED is letting to have a sync and at the same time utilize local KC password if it is manually entered for the user in admin console (or doing credential reset - send email). So idea is to catch new provisioned users from LDAP and put some required action to perform credentials reset that user could get a link to enter new password in KC and use it. 
Is it possible to achieve this by making some SPI or some other mechanism along build-in LDAP federation? Other options would be construct custom new LDAP sync SPI module or some other module eg utilizing Admin API.

**Comment by harish-moyal** [comment]
@gedpal - We have similar kind of requirement where we do want to sync users in LDAP but we want to bypass LDAP auth. Did you got any solution to your problem?

**Reply by gedpal** [reply]
Still in progress. Did some work creating custom LDAP mapper (SPI), catch new LDAP users creation events ( before it, failed to find mechanism to catch using  KC events system - didn't find any relevant event that could be created when new LDAP user is imported/created in KC UserDB) and fire some logic (eg create time limited link for the user to create new password and send it as email). If I finish this, probably we go on using it. Other options: create completely new LDAP sync module in KC or external module (free to choose Python or whatever) utilizing API calls.
